src/pager.rs

```diff
@@ -1,21 +1,52 @@
+//! Proivdes the [Pager] type
+
 use crate::{error::MinusError, input, minus_core::commands::Command, ExitStrategy, LineNumbers};
 use crossbeam_channel::{Receiver, Sender};
 use std::fmt;
 
 #[cfg(feature = "search")]
 use crate::search::SearchOpts;
 
-/// A pager acts as a middleman for communication between the main application
-/// and the user with the core functions of minus
+/// A communicationA bridge between the main application and the pager.
+///
+/// The [Pager] type which is a bridge between your application and running
+/// the running pager. Its the single most important type with which you will be interacting the
+/// most while working with minus. It allows you to send data, configure UI settings and also
+/// configure the key/mouse bindings.
 ///
-/// It consists of a [`crossbeam_channel::Sender`] and  [`crossbeam_channel::Receiver`]
+/// This neat thing about this is that not allows you to trasmits data while the pager is running
+/// but also if the pager hsen't been started. This means that you can preconfigure the pager
+/// before calling the initialization methods which most applications likely want to do.
+///
+/// It consists of a [`crossbeam_channel::Sender`] and [`crossbeam_channel::Receiver`]
 /// pair. When a method like [`set_text`](Pager::set_text) or [`push_str`](Pager::push_str)
-/// is called, the function takes the input. wraps it in the appropriate event
-/// type and transmits it through the sender held inside the this.
+/// is called, the function takes the input. wraps it in the appropriate command
+/// variant and transmits it through the sender held inside the this.
 ///
 /// The receiver part of the channel is continuously polled by the pager for events. Depending
-/// on the type of event that occurs, the pager will either redraw the screen or update
-/// the [PagerState](crate::state::PagerState)
+/// on the type of event that occurs, the pager will perform the required action.
+///
+/// [Pager] also implements the [std::fmt::Write] trait which means you can directly call [write!] and
+/// [writeln!] macros on it. For example, you can easily do this
+///
+/// ```
+/// use minus::Pager;
+/// use std::fmt::Write;
+///
+/// const WHO: &str = "World";
+/// let mut pager = Pager::new();
+/// writeln!(pager, "Hello {WHO}").unwrap();
+/// ```
+/// This appends `Hello World` to the end of minus's buffer. You can achieve the same using the
+/// [Pager::push_str] function like this
+/// ```
+/// use minus::Pager;
+/// use std::fmt::Write;
+///
+/// const WHO: &str = "World";
+/// let pager = Pager::new();
+/// pager.push_str(format!("Hello {WHO}\n")).unwrap();
+/// ```
```

Comment:
I'd replace this with something similar to this:
```md
/// The main struct of the library where you can interact and config
/// the pager.
/// 
/// # Example
/// ```
/// use minus::Pager;
/// use std::fmt::Write;
///
/// const WHO: &str = "World";
/// let mut pager = Pager::new();
///
/// // this...
/// writeln!(pager, "Hello {WHO}").unwrap();
///
/// // ... equals this.
/// pager.push_str(format!("Hello {WHO}\n")).unwrap();
/// ```
```
My reason is that someone who wants to use a library, he/she doesn't need to know how the library does XYZ internally. He/She just need to know what should be known when *using* it.

Reply:
I understand that you only want to show the usage docs but we also need to add dev docs for new contributors. Do we have an alternative way of writing that?

Reply:
> Do we have an alternative way of writing that?

For me, it's code. Seriously.
When I created my [PR for the following mode](https://github.com/arijit79/minus/pull/114) I was mostly spending my time by looking into the functions than reading the docs because they were often too big for me and I could gather more information through the code than through the docs. *Some* doc-notes were useful to get a better understanding but information like "we're using `tx` and `rx` to let the pager receive information" is a bit too much in my eyes.

*I hope that this message doesn't sound too unfriendly... please don't feel attacked, that's not my intention ;-;*

Reply:
Sorry I was busy for a while during the Christmas. In the meantime, I looked at several libraries that have a similar situation for docs and found that they have given a gentle introduction for how everything in the library is linked to each other in their lib.rs file and then leave rest as basic user docs. I think a similar pattern should also work for minus. What do you say?

Reply:
Can you check if its okay with my latest commit?

Reply:
I think it's fine, but I'd create a second review.